### ISSUE TITLE
question: what is it for to del a parameter in a function?

### ISSUE THREAD

**Issue by dandanmylady** [tensorflow/ranking#205]
## does 'del' a paramater really work as expected?
in ./python/losses_impl.py, method 'individual_weights' confused me a lot.  
in my opinion,  to del a parameter in a method actually doesn't del that object/tensor.  
Did i misunderstand anything?  

maybe this isn't the right place to ask this kind of question.  if there is any official mail list or forum, please tell me.  I'll very appreciate it.

### I tried with following simple python code.  del_obj() method really doesn't del the object.
```python
class MyClass(object):
    def __del__(self):
        print("obj is deleted")

def del_obj(obj):
    del obj
    

obj = MyClass()
del_obj(obj)
del obj
```

### following is the code of individual_weights
``` python
 def individual_weights(self, labels, ranks):
    """Returns the weight `Tensor` for individual examples.

    Args:
      labels: A dense `Tensor` of labels with shape [batch_size, list_size].
      ranks: A dense `Tensor` of ranks with the same shape as `labels` that are
        sorted by logits.

    Returns:
      A `Tensor` that can weight individual examples.
    """
    del ranks
    return labels
```

**Comment by ramakumar1729** [comment]
`del obj` is not the same as calling destructor of the `obj`. 

See [https://docs.python.org/2/reference/simple_stmts.html#the-del-statement](https://docs.python.org/2/reference/simple_stmts.html#the-del-statement).

"Deletion of a name removes the binding of that name from the local or global namespace, depending on whether the name occurs in a global statement in the same code block".

StackOverflow is more appropriate for these kinds of questions as we mainly try to address issues related to the ranking framework here. Feel free to reopen if you still consider this a relevant issue.

**Comment by dandanmylady** [comment]
Thank you so much @ramakumar1729  for your help!
So it is just to remove the binding of that name from local.  It just explicitly says that "ranker" tensor is not needed anymore in this function.   I got it.

Thanks again.

**Comment by ramakumar1729** [comment]
Yes, what you say makes sense. Thanks for your interest in tf-ranking, @dandanmylady !

**Comment by dandanmylady** [comment]
You are welcome, @ramakumar1729 
tf-ranking is an really helpful project.  
Without it, I/our team will  spend a lot of time to implement so many different kinds of ranking loss functions and there variances..
